app/models/streams/twitch_credentials.rb

```diff
@@ -0,0 +1,46 @@
+module Streams
+  class TwitchCredentials
+    ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY = :twitch_access_token_with_expiration
+
+    class << self
+      def access_token
+        token, exp = Rails.cache.fetch(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY)
+
+        if token.nil? || Time.zone.now >= exp
+          token, exp = get_new_token
+          Rails.cache.write(ACCESS_TOKEN_AND_EXPIRATION_CACHE_KEY, [token, exp])
+        end
+
+        token
+      end
```

Comment:
This uses the `Rails.cache` to store the tokens. This method also makes sure the token is not expired before returning it. If the token is expired, a new one is requested and returned instead (and added to the cache)